configure.ac

```diff
@@ -46,7 +44,27 @@ AC_ARG_WITH(extra-lib,
                           prefix where extra libraries are installed],
 [AC_SUBST(LDFLAGS, "$LDFLAGS -L${withval}")])
 
-CFLAGS_COMMON="-std=c++14 -Wall -fPIC -Wno-write-strings -I/usr/include/libnl3 -I/usr/include/swss"
+AC_ARG_WITH(extra-usr-lib,
+[  --with-extra-usr-lib=DIR
+                          prefix where extra libraries are installed],
+[AC_SUBST(LDFLAGS, "$LDFLAGS -L${withval}")])
+
+AC_CHECK_LIB([nl-3], [nl_addr_alloc])
```

Comment:
>nl_addr_alloc [](http://example.com/codeflow?start=22&length=13)

Since you are requiring a local patched libnl3, do you want to check the required new function `rtnl_route_nh_get_encap_mpls_dst`?

Reply:
updated.

```
checking for sai_object_type_query in -lsai... no                                                                                                                                    
configure: WARNING: libsai is not installed.                                                                                                                                         
checking for nl_addr_alloc in -lnl-3... yes                                                                                                                                          
checking for nl_socket_get_cb in -lnl-genl-3... yes                                                                                                                                  
checking for rtnl_route_nh_get_encap_mpls_dst in -lnl-route-3... yes                                                                                                                 
checking for nfnl_connect in -lnl-nf-3... yes                                                                                                                                        
checking that generated files are newer than configure... done                                                                                                                       

```